core/policy/src/main/java/orca/policy/core/ServiceManagerTicketReviewPolicy.java

```diff
@@ -0,0 +1,188 @@
+                                // but failed reservations don't contain an Authority Name
+
+                                // Find the site from type.
+                                // If e.g. a Slice has a failed VM at a site, we don't want to Ticket the VLAN there
+                                if (sliceReservation.getResources() != null && sliceReservation.getResources().getType() != null) {
+                                    String site = getSiteFromType(sliceReservation.getResources().getType().getType());
+
+                                    // compare site to list of site failures
+                                    if (!sliceFailureSites.containsKey(sliceID)) {
+                                        sliceFailureSites.put(sliceID, new ArrayList<String>());
+                                    }
+                                    List<String> failureSites = sliceFailureSites.get(sliceID);
+                                    if (!failureSites.contains(site)) {
+                                        failureSites.add(site);
+                                    }
+                                }
+                            }
+                        } else if (sliceReservation.getState() == ReservationStateNascent) {
+                            if (logger.isDebugEnabled()) {
+                                logger.debug("Found Nascent Reservation " +
+                                        sliceReservation.getReservationID() +
+                                        " in Slice " + slice.getName() +
+                                        " when checkPending() for " + reservation.getReservationID());
+                            }
+
+                            sliceStatusMap.put(sliceID, ReservationStateNascent);
+                        }
+                    }
+                }
+
+                // take action on the current reservation
+                if (sliceStatusMap.get(sliceID) == ReservationStateFailed) {
+                    if (reservation.getResources() != null && reservation.getResources().getType() != null) {
+                        // only fail the reservation if it from the same site as another failed reservation
+                        String site = getSiteFromType(reservation.getResources().getType().getType());
+                        if (sliceFailureSites.containsKey(sliceID) && sliceFailureSites.get(sliceID).contains(site)) {
+                            // Fail the reservation, and remove it from everything
+                            logger.info("TicketReview: Closing reservation " + reservation.getReservationID() +
+                                    " due to failure in Slice " + slice.getName());
+                            reservation.fail("TicketReview: Closing reservation due to failure in Slice.");
+                            actor.close(reservation); // "perform local close operations and issue close request to authority"
```

Comment:
Now that the SM intercepts and closes reservations, how does the controller react to it? I guess through querying on sliceStatus()?